temporalio/workflow.py

```diff
@@ -232,13 +279,12 @@ def with_name(
             )
         return fn
 
-    if name is not None or dynamic:
+    if not fn:
         if name is not None and dynamic:
             raise RuntimeError("Cannot provide name and dynamic boolean")
-        return partial(with_name, name)
-    if fn is None:
-        raise RuntimeError("Cannot create signal without function or name or dynamic")
```

Comment:
Not a big deal, but I think this protected against just invoking `@workflow.signal()` with no args (but not sure we need to protect against that, it'll just do nothing now instead of fail)